### ISSUE TITLE
if not send data to client, WSGIServer can not know the long polling connection has lost

### ISSUE THREAD

**Issue by zhangchunlin** [gevent/gevent#640]
For the long polling code of this url: http://sdiehl.github.io/gevent-tutorial/#long-polling
if there is "gevent.spawn(producer)" ,when the client has stop the connection, after a while when server yield a string to client, the server will get a exception of "error: [Errno 32] Broken pipe"

But if there is no "gevent.spawn(producer)" ,the server will not know the connection has lost at all.

The issue cause me can not do something immediately after client connection lost.

**Comment by jamadden** [comment]
This seems to be a duplicate of #88. Right now, there is no automatic, asynchronous way to get notified of a client disconnect. (Even at the TCP level, such notifications are not guaranteed to be immediate.) Most polling protocols (e.g., [socket.io](http://socket.io)) include the notion of periodic *heartbeat* messages to more quickly and reliably detect a disconnect.

**Comment by zhangchunlin** [comment]
@jamadden I don't think so, because if I send data with this connection the exception of "error: [Errno 32] Broken pipe" raise immediately , that is to say, gevent actually know the disconnection result before, but not raise it at the time gevent becoming to know it.

**Comment by zhangchunlin** [comment]
The disconnected action I tested is pressing the "stop" button of browser, so the client should close the connection normally, for this situation server side should know the disconnection, is it right?

**Comment by jamadden** [comment]
The broken pipe exception is raised at the first opportunity. This will be at a read or write call when the greenlet that's reading or writing is active. It will also be sometime after the operating system has detected the disconnection, which in turn will be sometime after the remote client has disconnected (from its standpoint). Note that it may take a trip around the libev event loop for the greenlet to be scheduled again and the exception propagated. 

In other words, there are no guarantees. It's all "eventually" and "best effort". If you never read or write from a socket, you'll never get an exception. If the greenlet that's using the socket is never scheduled, you'll never get an exception.
